plugin/metadata/setup.go

```diff
@@ -0,0 +1,49 @@
+package metadata
+
+import (
+	"github.com/coredns/coredns/core/dnsserver"
+	"github.com/coredns/coredns/plugin"
+	"github.com/mholt/caddy"
+)
+
+func init() {
+	caddy.RegisterPlugin("metadata", caddy.Plugin{
+		ServerType: "dns",
+		Action:     setup,
+	})
+}
+
+func setup(c *caddy.Controller) error {
+	m := &Metadata{}
+	c.Next()
+	zones := c.RemainingArgs()
```

Comment:
this should check the zones from the server block and use those as the default, potentially being overridden with these values, see other plugin for examples.

Also needs a setup_test.go